### PR TITLE
feat: added a new feature to copy all the job info

### LINKED ISSUE #9429: Troubleshoot Report
## Tell us about the problem you're trying to solve
As a user, I want to be able to download all of the troubleshooting information that the support team needs in order to help debug my problems. There are two cases that I want help debugging:
1. "general" platform error. Something went wrong while running airbyte (e.g. server error, scheduler error, some page is not loading, etc)
2. job error (e.g sync, spec, check, discover).

## Describe the solution you’d like
In either case, I want to click one button and be able to download all of the information that I need from the UI so that I can give it to a support engineer. 

The initial scope of this should to expose this in the API.

### PR DESCRIPTION
**Pull request by harshithmullapudi**
## What
Feature to let users copy the job info which matches with the issue template in troubleshooting. Closes https://github.com/airbytehq/airbyte/issues/9429

Also added more context to attempt details like (emitted/committed records) along with failure reason
closing https://github.com/airbytehq/airbyte/issues/10119

### CODE REVIEW

File: airbyte-webapp/src/components/JobItem/components/AttemptDetails.tsx [hunk 1]
```diff
@@ -61,36 +73,58 @@ const AttemptDetails: React.FC<IProps> = ({
...
+          <FormattedMessage
+            id="sources.countEmittedRecords"
+            values={{ count: attempt.totalStats?.recordsEmitted }}
+          />{" "}
           |{" "}
+        </span>
+        <span>
+          <FormattedMessage
+            id="sources.countCommittedRecords"
+            values={{ count: attempt.totalStats?.recordsCommitted }}
```
Comment (cgardens) [inline airbyte-webapp/src/components/JobItem/components/AttemptDetails.tsx:92]: sometimes recordsCommitted will be null. in those cases we should display the world unavailable.

**Review by cgardens — COMMENTED** [review]
Thank you!

@harshithmullapudi - Please add screenshots of the changes to the PR please so I can see what the changes look like.

Did the icons for partial success get updated? I didn't see them while reviewing.

A few suggestions on parts of the business logic that I did not explain clearly enough in the original issue. Otherwise, I think this gets us what we want!

I am only going to comment and let @Jamakase or @timroes be the approvers of the code itself.

**Comment by harshithmullapudi** [comment]
Screenshot 2022-02-06 at 4 12 38 AM
152662034-811d91
Screenshot 2022-02-06 at 4 36 46 AM
eb-39ad-477b-8114-216b1e320a12.png">
9-30c05b5d-f218-40f4-8ea8-b5d210a5e487.png">

Here are some screenshots.

File: airbyte-webapp/src/components/JobItem/components/AttemptDetails.tsx [hunk 2]
```diff
@@ -17,6 +17,18 @@ const Details = styled.div`
...
   color: ${({ theme }) => theme.greyColor40};
 `;
+const FailureReasonDetails = styled.div`
+  margin-top: 10px;
+`;
+
+const getFailureReason = (attempt: Attempt) => {
+  const {
+    failureSummary: { failures },
+  } = attempt;
```
[the remark this answers is not in the document]
Comment (cgardens) [reply airbyte-webapp/src/components/JobItem/components/AttemptDetails.tsx]: sorry. yeah. i meant `const failures = attempt.failureSummary`? I'm asking why you are doing the more complex variable assignment here.

**Comment by cgardens** [comment]
@harshithmullapudi thanks. Can you make sure we have the warning icon for the attempt tab as well, please?

File: airbyte-webapp/src/components/JobItem/components/AttemptDetails.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (harshithmullapudi) [reply airbyte-webapp/src/components/JobItem/components/AttemptDetails.tsx]: Nothing specific was just used to that way of writing code (as a practice).

**Review by timroes — CHANGES_REQUESTED** [review]
A general note to internationalization: When we want to format strings like "title: value" or "error (source)", we should not put it in code like `${i18n(title)}: ${i18n(value)}`. This makes an assumption, that every language would be able to separate key values pairs with a colon like this, or in the other example, that every language would use (this kind of) parantheses, which isn't the case for every language. Thus when concatenating strings like this, we should already put the overall concatenation as a i18n string, and the several parts would become arguments to this string. E.g. this should look as follows (I've inlined `defaultMessage` for bette readability here):

```ts
formatMessage(
    { id: "keyValuePair", defaultMessage: "{key}: {value}" },
    {
      key: formatMessage({ id: "sources.message" }),
      value: failureMessage,
    }
  );
```

Only that way we can ensure we're able to translate those phrases later properly into different languages.

File: airbyte-webapp/src/components/JobItem/components/AttemptDetails.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (timroes) [reply airbyte-webapp/src/components/JobItem/components/AttemptDetails.tsx]: I'd second that. It's more commonly in JS/TS to destructure variables like this. (I've even highlighted a couple of places in my review, where I think we should do it too :D)

File: airbyte-webapp/src/components/JobItem/components/DebugInfoDetailsModal.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  const getSourceDetails = () => {
+    return `${jobDebugInfo.sourceDefinition.name}(${jobDebugInfo.sourceDefinition.dockerImageTag})`;
+  };
+
+  const getDestinationDetails = () => {
+    return `${jobDebugInfo.destinationDefinition.name}(${jobDebugInfo.destinationDefinition.dockerImageTag})`;
+  };
+
+  const onCopyClick = () => {
+    navigator.clipboard.writeText(`${formatMessage({
```
[the remark this answers is not in the document]
Comment (harshithmullapudi) [reply airbyte-webapp/src/components/JobItem/components/DebugInfoDetailsModal.tsx:53]: Agreed.

**Comment by harshithmullapudi** [comment]
Hey @timroes made the requested changes. Kindly review when you get time

File: airbyte-webapp/src/core/domain/job/Job.ts [hunk 4]
```diff
@@ -22,6 +59,8 @@ export interface Attempt {
   endedAt: number;
   bytesSynced: number;
   recordsSynced: number;
+  totalStats: TotalStats;
+  failureSummary: FailedSummary;
```
Comment (timroes) [inline airbyte-webapp/src/core/domain/job/Job.ts:63]: Looking at the [specification](https://github.com/airbytehq/airbyte/blob/master/airbyte-api/src/main/openapi/config.yaml#L3196-L3199) both of those values are not required, and thus could be `undefined` (which should be reflected in the typing here).

File: airbyte-webapp/src/core/domain/job/JobsService.tsx [hunk 5]
```diff
@@ -31,6 +31,14 @@ class JobsService extends AirbyteRequestService {
 
     return job;
   }
+
+  public async getDebugInfo(jobId: string | number): Promise<any> {
+    const jobDebugInfo = await this.fetch<any>(`${this.url}/get_debug_info`, {
```
[the remark this answers is not in the document]
Comment (timroes) [reply airbyte-webapp/src/core/domain/job/JobsService.tsx:39]: We still use `any` here, even though we cast it later to a specific type. Let's please just make usage of the generic here properly and avoid carrying any `any` around here.

File: airbyte-webapp/src/core/domain/job/Job.ts [hunk 6]
```diff
@@ -38,6 +77,21 @@ export interface JobDetails {
   attempts: AttemptInfo[];
 }
 
+export interface JobDebugInfoMeta {
+  airbyteVersion: string;
+  id: number | string;
```
Comment (timroes) [inline airbyte-webapp/src/core/domain/job/Job.ts:82]: When can this be `string`? According to the specification this is of `type: integer`?

**Review by timroes — CHANGES_REQUESTED** [review]

File: airbyte-webapp/src/core/domain/job/Job.ts (hunk 4 above)
Comment (harshithmullapudi) [reply airbyte-webapp/src/core/domain/job/Job.ts:63]: I was searching on how to get this. I think would be better if we can have the same info on this page also https://airbyte-public-api-docs.s3.us-east-2.amazonaws.com/rapidoc-api-docs.html#post-/v1/workspaces/create.

Comment (harshithmullapudi) [reply airbyte-webapp/src/core/domain/job/Job.ts:63]: Oh yeah, that would be great.

**Review by timroes — APPROVED** [review]
Code LGTM, tested locally against OSS on Chrome (Linux). Some minor UI design concerns, but since we're anyway having a redesign of that replication history currently in the pipeline, I think we shouldn't bother with design tweaks in this PR.

**Pull request merged**